std/typecons.d

```diff
@@ -4175,8 +4216,7 @@ private static:
             {
                 preamble ~= "alias self = " ~ name ~ ";\n";
                 if (WITH_BASE_CLASS && !__traits(isAbstractFunction, func))
-                    //preamble ~= "alias super." ~ name ~ " parent;\n"; // [BUG 2540]
-                    preamble ~= "auto parent = &super." ~ name ~ ";\n";
+                    preamble ~= "alias parent = AliasSeq!(__traits(getMember, super, \"" ~ name ~ "\"))[0];";
```

Comment:
Yes, that's to work around a grammar limitation with `alias = __traits`. The compiler will complain something like "basic type expected, not __traits" and the identity template (which AliasSeq is) works around it, and the [0] kills the tupleness introduced by it.

I've used the pattern myself many times.